app/code/Magento/Catalog/Model/Product/Image.php

```diff
@@ -278,7 +278,7 @@ public function getQuality()
      */
     public function setKeepAspectRatio($keep)
     {
-        $this->_keepAspectRatio = (bool)$keep;
+        $this->_keepAspectRatio = $keep && $keep !== 'false';
```

Comment:
Actually, you could probably just to a json_decode($keep) here..

Reply:
Even if `json_decode` works it would be much slower and much less clear.

Reply:
Yeah, point well-taken 👍